src/Data/Nat/Bounded.agda

```diff
@@ -0,0 +1,69 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Bounded natural numbers, and their equivalence to `Fin`
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Nat.Bounded where
+
+open import Data.Nat.Base as ℕ
+import Data.Nat.Properties as ℕ
+open import Data.Fin.Base as Fin using (Fin; zero; suc; toℕ)
+import Data.Fin.Properties as Fin
+open import Function.Base using (id; _∘_; _on_)
+open import Relation.Binary.Core using (_⇒_)
+open import Relation.Binary.PropositionalEquality.Core as ≡ using (_≡_; refl)
+open import Relation.Nullary.Decidable.Core using (recompute)
+open import Relation.Nullary.Negation.Core using (¬_)
+
+private
+  variable
+    m n : ℕ
+
+------------------------------------------------------------------------
+-- Definition
+
+record BoundedNat (n : ℕ) : Set where
+  constructor ⟦_⟧<
+  field
+    value : ℕ
+    .{{bounded}} : value < n
+
+open BoundedNat public using () renaming (value to ⟦_⟧)
+
+-- Constructor
+
+⟦0⟧< : ∀ n → .{{ NonZero n }} → BoundedNat n
+⟦0⟧< (n@(suc _)) = ⟦ zero ⟧< where instance _ = z<s
+
+-- Destructors
+
+¬BoundedNat[0] : ¬ BoundedNat 0
+¬BoundedNat[0] ()
+
+isBounded : (i : BoundedNat n) → ⟦ i ⟧ < n
+isBounded (⟦ _ ⟧< {{bounded}}) = recompute (_ ℕ.<? _) bounded
+
+-- Equality on values is propositional equality
+
+bounded≡⇒⟦⟧≡⟦⟧ : _≡_ {A = BoundedNat n} ⇒ (_≡_ on ⟦_⟧)
+bounded≡⇒⟦⟧≡⟦⟧ = ≡.cong ⟦_⟧
+
+⟦⟧≡⟦⟧⇒bounded≡ :  (_≡_ on ⟦_⟧) ⇒ _≡_ {A = BoundedNat n}
+⟦⟧≡⟦⟧⇒bounded≡ refl = refl
+
+------------------------------------------------------------------------
+-- Conversion to and from `Fin n`
```

Comment:
Surprised you didn't prove the full iso?

Reply:
I can be very lazy... and I haven't had much experience in `stdlib` of proving isomorphisms (esp. at the `Set`/`Setoid` level) as really adding value...

... so, I'd rather leave that (to somebody else?) as a supplementary PR?

Reply:
And to the last point: worth raising a PR to add something to `Function.Consequences`?

Reply:
I don't go through `Bijective` myself, I tend to go for `Equivalence`. The latter is more categorical while the former is more set-theoretical.

Reply:
And now that I look at the code... you actually prove `Equivalence` and don't write it down explicitly, and then proceed to "throw it away" to get `Bijective` !!! Wow.

Reply:
Oh, I see now...